tests/join-metadata-and-clades/join-metadata-and-clades.t

```diff
@@ -0,0 +1,25 @@
+Join pathogen metadata pulled from an external database with nextclade clade calls into one final metadata file.
+
+Create metadata file for testing.
+
+  $ cat > metadata_raw.tsv <<~~
+  > strain	date
+  > id_1	2023-01-01
+  > id_2	2023-02-02
+  > ~~
+
+Create nextclade file for testing.
+
+  $ cat > nextclade.tsv <<~~
+  > seqName	clade	outbreak	lineage	coverage	totalMissing	totalSubstitutions	totalNonACGTNs	qc.missingData.status	qc.mixedSites.status	qc.privateMutations.status	qc.frameShifts.status	qc.stopCodons.status	frameShifts	isReverseComplement
+  > id_1	val_1	val_1	val_1	val_1	val_1	val_1	val_1	val_1	val_1	val_1	val_1	val_1	val_1	val_1
+  > id_2	val_2	val_2	val_2	val_2	val_2	val_2	val_2	val_2	val_2	val_2	val_2	val_2	val_2	val_2
+  > ~~
+
+Check whether join-metadata-clades script produces an output metadata file, but do not assess the accuracy or validity of that output file.
+
+  $ python $TESTDIR/../../join-metadata-and-clades.py \
+  >  --id-field strain \
+  >  --metadata metadata_raw.tsv \
+  >  --nextclade nextclade.tsv \
+  >  -o test_metadata.tsv
```

Comment:
We should do at least some minor checking of the output file here. I think at minimum we want to ensure that all of the sequences in the metadata file are included in the output.

Reply:
Added a minor check in a5fa32ef0e924e7696904cb9f26259ed7cad7567
I'm happy to discuss a more detailed checking strategy; I was unsure on level of detail.

Reply:
General strategy is just to ensure that the output has the expected format. If we wanted cover all of our bases, we could include an expected output file and just diff the output with the expected output file.